### PR TITLE
Docs and readme fixes

### PR DESCRIPTION
**Pull request by ItayGabbay**
#### Reference Issues/PRs

None.

#### What does this implement/fix? Explain your changes.
- Fixing some issues in the docs generation
- Fixing links between notebooks and API reference.
- Fixing Colab & Binder links
- Fixing readme references to the docs

#### Any other comments?

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

**Review by noamzbr — APPROVED** [review]

File: docs/source/examples/index.rst [hunk 1]
```diff
@@ -1,22 +1,25 @@
 Examples
 ========
 
+Guides
```
Comment (shir22) [inline docs/source/examples/index.rst:4]: How-to Guides

File: docs/source/user-guide/key_concepts.rst [hunk 2]
```diff
@@ -9,7 +9,7 @@ Check
 
 Each check enables you to inspect a specific aspect of your data and models.
 They are the basic building block of the deepchecks package, covering all kinds of common issues,
-such as: PerformanceOverfit, DataSampleLeakage, SingleFeatureContribution, DataDuplicates, and :ref:`many more checks <examples/checks>`.
+such as: PerformanceOverfit, DataSampleLeakage, SingleFeatureContribution, DataDuplicates, and :ref:`many more checks <examples-check-demonstrations>`.
```
Comment (shir22) [inline docs/source/user-guide/key_concepts.rst:12]: why is this the link?
isn't it still in examples/checks...?

Comment (ItayGabbay) [reply docs/source/user-guide/key_concepts.rst:12]: It's a label

Comment (shir22) [reply docs/source/user-guide/key_concepts.rst:12]: or actually i guess it should be to the :doc: directive
(or to a direct label that you added? if so then there is a general convention for them, but I guess doc directive is the best)

**Pull request merged**